core/src/main/java/io/parsingdata/metal/data/ParseGraph.java

```diff
@@ -118,4 +119,20 @@ public String toString() {
         return "graph(" + head + ", " + tail + ", " + branched + ")";
     }
 
+    @Override
+    public boolean equals(Object obj) {
+        return obj != null
+            && getClass() == obj.getClass()
+            && Objects.equals(head, ((ParseGraph)obj).head)
+            && Objects.equals(tail, ((ParseGraph)obj).tail)
+            && Objects.equals(branched, ((ParseGraph)obj).branched)
+            && Objects.equals(definition, ((ParseGraph)obj).definition);
```

Comment:
I see your point but I'm not convinced. I propose we keep this for now and see if we want to change it later.